duckduckgo/iOS
# [DuckPlayer] - 4. Remote Config

Task/Issue URL: https://app.asana.com/0/1204099484721401/1207593922684293/f

**Description**:
- Adds a remote configuration flag to disable Duck Player
- Move Notifications to DuckPlayerSettings (for both Feature flag and NotificationCenter)
- Moves DuckPlayerSettings class to its own file
- Update UserScripts to initialize DuckPlayer scripts automatically
- Add protocols to DuckPlayer Classes for easier testing and improve tests




**Steps to test this PR**:
1. Enable DP via loading this Privacy Config: https://www.jsonblob.com/api/1252645298605252608
2. Smoke test DuckPlayer
3. Change DuckPlayer to be in Always ask mode
4. Open a Youtube video and confirm you see the overlay
5. Without changing the URL, Disable DP by loading this Privacy Config: to https://jsonblob.com/api/1257316236932276224
6. Right after you load it, the video will start playing.
7. Close settings, and you'll not see the overlay anyumore
8. Navigate to a duck:// URL, like:   duck://player/9vhR1RnMbSA
9. Confirm you're redirected to the Youtube video.
